source/llassetgen/include/llassetgen/Packing.h

```diff
@@ -1,173 +1,10 @@
-     *   a ForwardIterator, i.e. allow iterating over the input multiple times.
-     * @param[in] sizesEnd
-     *   End iterator for the rectangle sizes. Same constraints as for
-     *   `sizesBegin` apply.
-     * @param[in] allowRotations
-     *   Whether to allow rotating rectangles by 90˚.
-     * @return
-     *   Resulting packing.
-     */
-    template <class ForwardIter>
-    Packing shelfPackAtlas(ForwardIter sizesBegin, ForwardIter sizesEnd, bool allowRotations) {
-        internal::assertCorrectIteratorType<ForwardIter>();
-
-        auto rectCount = static_cast<size_t>(std::distance(sizesBegin, sizesEnd));
-        Packing packing{internal::predictAtlasSize(sizesBegin, sizesEnd)};
-        packing.rects.reserve(rectCount);
-
-        internal::ShelfNextFitPacker packer{packing, allowRotations, true};
-        bool success = std::all_of(sizesBegin, sizesEnd,
-                                   [&](Vec2<PackingSizeType> rectSize) { return packer.packNext(rectSize); });
-        // Only happens when a rectangle is wider (and higher, if rotations are enabled)
-        // than the width of the atlas texture, however this should be prevented
-        // by predictAtlasSize.
-        assert(success);
-
-        return packing;
-    }
-
-    /**
-     * Use the shelf next fit algorithm to pack a fixed size texture atlas.
-     *
-     * This algorithm can produce suboptimal packings, but has very fast runtime
-     * (O(n), where n is the number of rectangles to pack).
-     *
-     * @param[in] sizesBegin
-     *   Begin iterator for the sizes of the input rectangles. This iterators
-     *   items must be convertible to `Vec2<PackingSizeType>`. It must also be
-     *   a ForwardIterator, i.e. allow iterating over the input multiple times.
-     * @param[in] sizesEnd
-     *   End iterator for the rectangle sizes. Same constraints as for
-     *   `sizesBegin` apply.
-     * @param[in] allowRotations
-     *   Whether to allow rotating rectangles by 90˚.
-     * @param[out] packing
-     *   Instance to store the packing into. The atlas size should already be
-     *   set to the desired value. The `rects` vector must be empty, but may
-     *   have an appropriate amount of memory reserved to improve performance.
-     *   If the packing fails, this object is left in an unfinished state.
-     * @return
-     *   Whether the rectangles fit inside the atlas
-     */
-    template <class ForwardIter>
-    bool shelfPackFixedSizeAtlas(ForwardIter sizesBegin, ForwardIter sizesEnd, bool allowRotations, Packing& packing) {
-        internal::assertCorrectIteratorType<ForwardIter>();
-
-        internal::ShelfNextFitPacker packer{packing, allowRotations, false};
-        return std::all_of(sizesBegin, sizesEnd,
-                           [&](Vec2<PackingSizeType> rectSize) { return packer.packNext(rectSize); });
-    }
-}
+/** @file
+ * Used to include all packing headers at once.
```

Comment:
So, why is it needed now? To hide that Packing always needs Algorithms and Types? User convenience? (just asking)

Reply:
Just for convenience (only one include needed). The only reason that the public headers needed to be split in two was to break circular references between the public and internal headers, and I wanted to preserve the original header as an "include everything" option.

Reply:
That's seems a pretty good idea to me.